runtime/src/precompiles.rs

```diff
@@ -0,0 +1,71 @@
+use pallet_evm::{
+    IsPrecompileResult, Precompile, PrecompileHandle, PrecompileResult, PrecompileSet,
+};
+use sp_core::H160;
+use sp_std::marker::PhantomData;
+
+use pallet_evm_precompile_modexp::Modexp;
+use pallet_evm_precompile_sha3fips::Sha3FIPS256;
+use pallet_evm_precompile_simple::{ECRecover, ECRecoverPublicKey, Identity, Ripemd160, Sha256};
+
+pub struct FrontierPrecompiles<R>(PhantomData<R>);
+
+impl<R> FrontierPrecompiles<R>
+where
+    R: pallet_evm::Config,
+{
+    pub fn new() -> Self {
+        Self(Default::default())
+    }
+    pub fn used_addresses() -> [H160; 7] {
+        [
+            hash(1),
+            hash(2),
+            hash(3),
+            hash(4),
+            hash(5),
+            hash(1024),
+            hash(1025),
+        ]
+    }
+}
+
+impl<R> Default for FrontierPrecompiles<R>
+where
+    R: pallet_evm::Config,
+{
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl<R> PrecompileSet for FrontierPrecompiles<R>
+where
+    R: pallet_evm::Config,
+{
+    fn execute(&self, handle: &mut impl PrecompileHandle) -> Option<PrecompileResult> {
+        match handle.code_address() {
+            // Ethereum precompiles :
+            a if a == hash(1) => Some(ECRecover::execute(handle)),
+            a if a == hash(2) => Some(Sha256::execute(handle)),
+            a if a == hash(3) => Some(Ripemd160::execute(handle)),
+            a if a == hash(4) => Some(Identity::execute(handle)),
+            a if a == hash(5) => Some(Modexp::execute(handle)),
+            // Non-Frontier specific nor Ethereum precompiles :
+            a if a == hash(1024) => Some(Sha3FIPS256::execute(handle)),
+            a if a == hash(1025) => Some(ECRecoverPublicKey::execute(handle)),
+            _ => None,
```

Comment:
Should we include `Dispatch` for extra shenanigans?

Reply:
Good question. For those who don't know dispatch is an evm precompile that allows contracts to dispatch arbitrary runtime extrinsics. One implication in this chain is that it would mean evm contracts could dispatch into pallet contracts.

I don't know whether it is possible to enable the same thing from wasm to evm or not, but that would really make things crazy. If we do enable dispatch, do it in a followup.

Reply:
So I'm thinking for the sake of this PR, no. If you want to try that as a followup, I'm open to considering it. Maybe focus on the lecture first :sweat_smile: